File: android/Girlsfleet/app/src/main/java/com/circus/girlsfleet/IGameSystem.java
package com.circus.girlsfleet;

import android.content.res.AssetManager;

import java.util.Random;
import java.util.Stack;

public interface IGameSystem {

    ScreenMode getScreenMode();
    Stack<IGameState> getStateStack();
    Random getRandom();
    AssetManager getAssetManager();

}

File: android/Circus/app/src/main/java/com/personal/circus/IGameSystem.kt
package com.personal.circus;
import android.content.res.AssetManager
import java.util.*;
import kotlin.collections.HashMap

interface IGameSystem {
    fun getScreenMode():ScreenMode;
    fun getStateStack():Stack<IGameState>;
    fun getRand():Random;

    fun getAssets():AssetManager

}